### PR TITLE
Implemented rearrange option in section.

### PR DESCRIPTION
**Pull request by Parrsifal**
Implemented edit button in navigation bar and ability to enter in edit mode.

### CODE REVIEW

File: ToDoList/ToDoList/Modules/TaskManagerModule/View/TaskListViewController.swift [hunk 1]
```diff
@@ -36,12 +36,24 @@ final class TaskListViewController: UIViewController, TaskListView {
...
+//    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+//     super.traitCollectionDidChange(previousTraitCollection)
+//             if traitCollection.hasDifferentColorAppearance(comparedTo: previousTraitCollection) {
+////            self.view.setNeedsLayout()
+////            self.view.setNeedsDisplay()
+//            //self.view.layoutIfNeeded()
+//            taskListTableView.reloadData()
+//
+//           }
+//    }
```
Comment (niclupu) [inline ToDoList/ToDoList/Modules/TaskManagerModule/View/TaskListViewController.swift:55]: Do we need this commented code?

File: ToDoList/ToDoList/Modules/AddTaskModule/View/AddTaskViewController.swift [hunk 2]
```diff
@@ -45,9 +45,9 @@ final class AddTaskViewController: UIViewController, AddTaskView {
     }
     
     private func setUpScreenMode() {
-        presenter.setUpScreenMode(title: titleTextField.text ?? "", description: descriptionTextField.text)
+       presenter.setUpScreenMode(title: titleTextField.text ?? "", description: descriptionTextField.text)
```
Comment (niclupu) [inline ToDoList/ToDoList/Modules/AddTaskModule/View/AddTaskViewController.swift:48]: didn't notice this earlier, but I see there's no need for these parameters, as you give the title and subtitle from task in presenter

File: ToDoList/ToDoList/Modules/TaskManagerModule/View/TaskListViewController.swift [hunk 3]
```diff
@@ -36,12 +36,24 @@ final class TaskListViewController: UIViewController, TaskListView {
         setUpNavBar()
         setUpTableView()
         setUpBackButton()
-        
+        setUpEditButton()
     }
     
     override func viewWillAppear(_ animated: Bool) {
         taskListTableView.reloadData()
```
Comment (niclupu) [inline ToDoList/ToDoList/Modules/TaskManagerModule/View/TaskListViewController.swift:43]: let's better have a method available through view protocol that set's the tasks and updates the tableView. Here we'll just notify presenter that view will appear, and it is gonna get the tasks from storage and then update the view.
```
func updateTaskList(tasks: [Task]) {
   self.tasksList = tasksList
   taskListTableView.reloadData()
}
```
In presenter you'll call this method whenever you need to update the tableview

File: ToDoList/ToDoList/Modules/AddTaskModule/View/AddTaskViewController.swift [hunk 4]
```diff
@@ -45,17 +45,17 @@ final class AddTaskViewController: UIViewController, AddTaskView {
     }
     
     private func setUpScreenMode() {
```
Comment (vikabelinschi) [inline ToDoList/ToDoList/Modules/AddTaskModule/View/AddTaskViewController.swift:47]: A good practice is to have a structure for your classes. Usually, we keep the private methods at the end of our class, and the internal and overriden ones at the top.

File: ToDoList/ToDoList/Modules/AddTaskModule/View/AddTaskViewController.swift [hunk 5]
```diff
@@ -35,51 +35,44 @@ final class AddTaskViewController: UIViewController, AddTaskView {
...
-    func reloadStatus() {
-        titleTextField.text = ""
-        descriptionTextField.text = ""
-        hideKeyboard()
-        createTaskButton.alpha = 0
+    @IBAction private func inputFieldsChanged(_ sender: UITextField) {
+        presenter.inputFieldsWasChanged(titleTextField.text!, descriptionTextField.text)
     }
     
+    @IBAction private func buttonDidTouch(_ sender: UIButton) {
```
Comment (niclupu) [inline ToDoList/ToDoList/Modules/AddTaskModule/View/AddTaskViewController.swift:67]: I would rename this method buttonWasTapped, as buttonDidTouch may seem like the button touched something )))

**Review by niclupu — APPROVED** [review]

**Pull request merged**